### ISSUE TITLE
handleActions naming is confusing

### ISSUE THREAD

**Issue by dapetcu21** [acdlite/redux-actions#47]
Since `handleActions` returns a reducer, a proper name for it would be `createReducer`. I've tried to teach redux to a friend and it wasn't immediately apparent that `handleActions` returns a reducer. What do you think? I'm not adamant on `createReducer`. I just think that `handleActions` might not be the proper name for it.

**Comment by acdlite** [comment]
Hmm, I see what you mean but `createReducer()` may be generic for my taste. Perhaps `reducerForActions()`. I think `handleActions()` is fine but I'm open to other suggestions :)

**Comment by dapetcu21** [comment]
Yeah, `reducerForAction` and `reducerForActions` sounds more reasonable.

**Comment by shark0der** [comment]
I'd add on this.
Let's take this code straight from the documentation:

`let increment = createAction('INCREMENT');`

Any person that would see this for the first time would assume that `increment` is now an action. Well... bad news, `increment` is now an action therefore an actually descriptive name for `createAction` in this case would be `createActionCreator`, which reminds me of this: http://bash.org/?946461.

Please come up with some proper names :disappointed:.

**Comment by doliver3** [comment]
Recommendations:
A better name for createAction is **actionCreator** since it generates an actionCreator
A better name for handleActions is **actionReducers** since it generates reducers.

This lines up with existing naming conventions:
http://redux.js.org/docs/basics/Actions.html - Action Creators
http://redux.js.org/docs/basics/Reducers.html - Reducers handle Actions

The react-redux-starter-kit just added the 'redux-actions' names to its module/counter file as it turned 1.0 recently, and as someone new to Redux I would agree that the naming of 'redux-actions' was not intuitive and it was significantly different than the Redux example projects. Only the comments made it clear what each was.

**Comment by rpedroni** [comment]
I can totally back this up. Even though I understand quite well redux terminology, I've been confused a couple of times by the used names.
If at least `createAction` could be renamed to `buildActionCreator` or even `createActionCreator`, It'd stick to the correct redux terms and be way easier to pick up.

But not to just be a bitch about it, thank you for this lib :)

**Comment by ntucker** [comment]
I agree with naming of @rpedroni . Another possibility is 'make'. aka, makeActionCreator, makeActionReducer

**Comment by acornejo** [comment]
just started using this library (great btw!), and I was also tripped up by the naming. of course, its easy to figure out after using it a bit, but I agree that renaming these would reduce friction for new users.

**Comment by anyong** [comment]
+1 `makeActionCreator` and `makeActionReducer`

**Comment by yangmillstheory** [comment]
From this thread, it sounds like people would prefer the following (or a slight variant of the following)

- `makeActionCreators`
- `makeActionCreator`
- `makeActionReducers`
- `makeActionReducer`

Dropping the `make` prefix would make it a bit less verbose.

**Comment by Kadrian** [comment]
I would personally favor `makeReducer` over `makeActionReducer`, since people who know redux, typically know that a reducer reduces actions. 
How about:

* `makeActionCreator`
* `makeActionCreators`
* `makeReducer`
* `makeReducers`

Is there currently someone taking care of this / will this be part of a milestone soon?

**Comment by yangmillstheory** [comment]
Yes, I can take care of this (it's actually trivial and I've done it in a fork). 

We could release it as `v1.2.0` and alias the deprecated names (which log warnings when used), then completely remove it in `v2.0.0`.

Relatedly, are the singular forms of these methods useful anymore? The only reason I think it's worthwhile keeping them is backwards-compatibility, and it feels like it's better to just keep the plural forms, since it'd reduce API surface area.

**Comment by codeaid** [comment]
What about replacing `make` with `get`? 

- `getActionCreator`
- `getActionCreators`
- `getReducer`
- `getReducers`

**Comment by mslipper** [comment]
@yangmillstheory did you ever finish this?

**Comment by yangmillstheory** [comment]
No (take a look at the code).

**Comment by yangmillstheory** [comment]
No (take a look at the code).

**Comment by mslipper** [comment]
👍 , just wanted to make sure that it wasn't sitting in a branch somewhere. I can take this.

**Comment by malyzeli** [comment]
@yangmillstheory what's the progress? 

If it's just about renaming with aliases and adding deprecation warning it's really a simple task doable in couple of minutes. 

Since I'm actually renaming it in my project (importing `redux-actions` code by single utility file and exporting renamed functions for rest of the code), I wouldn't mind sending a pull request here. 

I will do it if somebody is gonna merge it, just let me know.

Personally I stand for explicit naming even though it is more verbose, so I use `createActionCreator` form. If you want shorter code, you can always alias it at the import level...
